source/simulator/assemblers/newton_stokes.cc

```diff
@@ -328,21 +416,23 @@ namespace aspect
           // first assemble the rhs
           for (unsigned int i=0; i<stokes_dofs_per_cell; ++i)
             {
-              data.local_rhs(i) -= (eta * 2.0 * (scratch.grads_phi_u[i] * strain_rate)
+              data.local_rhs(i) -= (eta * 2.0 * (scratch.grads_phi_u[i] *
+                                                 scratch.material_model_inputs.strain_rate[q])
```

Comment:
Should this be the `effective_strain_rate`?

Reply:
No, it must be the original strain_rate without deviator() here, because when $u$ is the true solution then the RHS equals 0, a deviator() only breaks the equivalence and results in slow convergence.

Reply:
Makes sense, thanks!